storage/client.go

```diff
@@ -437,12 +437,12 @@ func (c Client) exec(verb, url string, headers map[string]string, body io.Reader
 			// response contains storage service error object, unmarshal
 			if resp.Header.Get("Content-Type") == "application/xml" {
 				errIn := serviceErrFromXML(respBody, &storageErr)
-				if err != nil { // error unmarshaling the error response
+				if errIn != nil { // error unmarshaling the error response
 					err = errIn
```

Comment:
Good catch seeing the masked error. However, I'm not sure this fix goes far enough to fix it. As it is in this PR, `errIn` gets blown away anyway.

@mcardosos, what are your thoughts? Maybe we should be extending the AzureStorageServiceError to include an inner error in the case of marshaling trouble?

Reply:
Take a look into the `serviceErrFromXML` and `serviceErrFromJSON` funcs. They both include the unmarshal errors in `AzureStorageServiceError.Message`. I actually think we could get rid of `if err != nil`. Code would look like this...
```go
if resp.Header.Get("Content-Type") == "application/xml" {
	serviceErrFromXML(respBody, &storageErr)
} else {
	serviceErrFromJSON(respBody, &storageErr)
}
```

Reply:
Talking IRL with @mcardosos, she showed me that we do actually percolate up the un-marshaling error already. I like the blurb that she has above, let's do that :)

Reply:
@marstr I noticed afterwards that the error is blown away, but thought that you need to take a look at the whole code block. As I did this at work I didn't want to put too much time into it.